pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -413,27 +413,27 @@ func buildAutoMtlsSettings(
 	meshExternal bool,
 	serviceMTLSMode model.MutualTLSMode) (*networking.ClientTLSSettings, mtlsContextType) {
 	if tls != nil {
+		if tls.Mode == networking.ClientTLSSettings_DISABLE || tls.Mode == networking.ClientTLSSettings_SIMPLE {
+			return tls, userSupplied
+		}
+		subjectAltNamesToUse := tls.SubjectAltNames
```

Comment:
Curious - any reason for moving this before sni ?

Reply:
There is no specific reason